File: src/java/org/javaturk/dp/ch05/facade/computer/RAM.java
package org.javaturk.dp.ch05.facade.computer;

public class RAM {
	private BIOS bios;
	private boolean kernelLoaded = false;
	private boolean osLoaded = false;

	public RAM() {
		bios = new BIOS();
		double random = Math.random();
		if (random < 0.8) {
			kernelLoaded = true;
			osLoaded = true;
		}
	}

	public boolean loadOSKernelImage() {
		System.out.println("RAM: Loading the image of OS kernel.");
		if (kernelLoaded) {
			bios.start();
			System.out.println("RAM: Image of OS kernel has been loaded.");
		} else
			System.out.println("RAM: Image of OS kernel has been loaded.");
		return kernelLoaded;
	}

	public boolean loadOS() {
		if (osLoaded)
			System.out.println("OS has been loaded successfully.");
		else
			System.out.println("OS loading has been failed.");
		return osLoaded;
	}

	public void loadDrivers() {
		System.out.println("Drivers have been loaded.");
	}

	public void startServices() {
		System.out.println("Services have started.");
	}
	
	public void fixIt(){
		System.out.println("Fixing kernel and OS!");
		kernelLoaded = true;
		osLoaded = true;
	}
}

File: src/typescript/com/gokselkucuksahin/dp/creational-patterns/builder/computer/domain/ram.ts
import {BIOS} from "./bios";

export class RAM {
  private bios: BIOS;

  constructor() {
    this.bios = new BIOS();
  }

  loadOSKernelImage(): void {
    console.log("RAM: Loading the image of OS kernel.");
    this.bios.start();
    console.log("RAM: Image of OS kernel has been loaded.");
  }

  loadOS(): void {
    console.log("OS has been loaded.");
  }

  loadDrivers(): void {
    console.log("Drivers have been loaded.");
  }

  startServices(): void {
    console.log("Services have started.");
  }
}